tests/test_boto3_client.py

```diff
@@ -0,0 +1,127 @@
+from unittest.mock import call
+from unittest.mock import patch
+
+from common.object_storage_adaptor.boto3_client import Boto3_Client
+from common.object_storage_adaptor.boto3_client import get_boto3_client
+from tests.conftest import PROJECT_CREDENTIALS
+
+
+@patch('aioboto3.Session')
+async def test_boto3_client_init_connection_with_token_requests_credentials_creates_boto3_session(
+    _session, mock_post_by_token
+):
+    boto3_client = Boto3_Client(endpoint='project', token='test')
+    await boto3_client.init_connection()
+
+    # Asserting that boto3 session was created with correct credentials
+    _session.assert_called_with(
+        aws_access_key_id=PROJECT_CREDENTIALS.get('AccessKeyId'),
+        aws_secret_access_key=PROJECT_CREDENTIALS.get('SecretAccessKey'),
+        aws_session_token=PROJECT_CREDENTIALS.get('SessionToken'),
+    )
+    # Asserting that boto3_client have a mocked boto3 session
+    assert boto3_client._session == _session()
+
+
+@patch('aioboto3.Session')
+async def test_get_boto3_client(_session):
+    boto3_client = await get_boto3_client(endpoint='project', temp_credentials=PROJECT_CREDENTIALS)
+
+    # Asserting that we get a correct boto3 client class
+    assert isinstance(boto3_client, Boto3_Client)
+    # Asserting that boto3 session was created with correct credentials
+    _session.assert_called_with(
+        aws_access_key_id=PROJECT_CREDENTIALS.get('AccessKeyId'),
+        aws_secret_access_key=PROJECT_CREDENTIALS.get('SecretAccessKey'),
+        aws_session_token=PROJECT_CREDENTIALS.get('SessionToken'),
+    )
+
+
+@patch('aioboto3.Session.client')
+async def test_boto3_client_downloads_the_file_from_s3(_client):
+    boto3_client = Boto3_Client(endpoint='project', temp_credentials=PROJECT_CREDENTIALS)
+    await boto3_client.init_connection()
+    args = ('test', '/test/path', '/home/test/path')
+    await boto3_client.download_object(*args)
+
+    # Asserting that boto3 download_file method gets called with correct args
+    assert _client.call_count == 1
+    _client.assert_has_calls(
+        [
+            call().__aenter__().download_file(*args),
+        ]
+    )
+
+
+@patch('aioboto3.Session.client')
+async def test_boto3_client_copy_file_copies_the_file_from_s3(_client):
+    boto3_client = Boto3_Client(endpoint='project', temp_credentials=PROJECT_CREDENTIALS)
+    await boto3_client.init_connection()
+    await boto3_client.copy_object('test', '/test/path', '/test/path/new')
+
+    # Asserting that boto3 copy_file method gets called with correct kwargs
```

Comment:
If you're going to leave these comments, please also fix this.
```suggestion
    # Asserting that boto3 copy_object method gets called with correct kwargs
```